crates/cairo-annotations/src/annotations/profiler.rs

```diff
@@ -29,12 +31,46 @@ pub struct ProfilerAnnotationsV1 {
     pub statements_functions: HashMap<StatementIdx, Vec<FunctionName>>,
 }
 
+lazy_static! {
+    static ref RE_LOOP_FUNC: Regex = Regex::new(r"\[expr\d*\]")
+        .expect("Failed to create regex normalising loop functions names");
+    static ref RE_MONOMORPHIZATION: Regex = Regex::new(r"<.*>")
+        .expect("Failed to create regex normalising mononorphised generic functions names");
+}
+
 /// The fully qualified Cairo path of the Cairo function.
 #[derive(
     Clone, Debug, Eq, PartialEq, Hash, Ord, PartialOrd, Serialize, Deserialize, Display, Default,
 )]
 pub struct FunctionName(pub String);
 
+impl FunctionName {
+    #[must_use]
+    pub fn from_sierra_statement_idx(
+        statement_idx: StatementIdx,
+        sierra_program: &Program,
+        split_generics: bool,
+    ) -> Self {
+        // The `-1` here can't cause an underflow as the statement id of first function's entrypoint is
+        // always 0, so it is always on the left side of the partition, thus the partition index is > 0.
+        let function_idx = sierra_program
+            .funcs
+            .partition_point(|f| f.entry_point.0 <= statement_idx.0)
+            - 1;
+        let function_name = sierra_program.funcs[function_idx].id.to_string();
+        // Remove suffix in case of loop function e.g. `[expr36]`.
+        let function_name = RE_LOOP_FUNC.replace(&function_name, "");
+        // Remove parameters from monomorphised Cairo generics e.g. `<felt252>`.
+        let function_name = if split_generics {
+            function_name
+        } else {
+            RE_MONOMORPHIZATION.replace(&function_name, "")
```

Comment:
Maybe let's move this to another function like this:

```rust
    /// Remove parameters from monomorphised Cairo generics e.g. `<felt252>`.
    pub fn remove_generics(&self) -> Self {
        let function_name = RE_MONOMORPHIZATION.replace(&self.0, "");
        Self(function_name.to_string())
    }
```

As this could be useful at any part of lifetime of `FunctionName`

Then the signature could be change to this

```rust
  pub fn from_sierra_statement_idx(
        statement_idx: StatementIdx,
        sierra_program: &Program,
    ) -> Self
```

Reply:
Not sure if this would be better, for a few reasons:

- the if there has pretty low complexity, and by moving it out we loose context on when it should be used
- the reusability of such function is actually very low, but we increase indirection and make it just a bit harder to just use this
- it also requires me to have additional if in profiler :( 

Any strong arguments (or actual use cases) for having this externalized?

Reply:
You convinced me, let's leave it as it is